packages/animations/lib/src/open_container.dart

```diff
@@ -535,6 +578,51 @@ class _OpenContainerRoute extends ModalRoute<void> {
     Animation<double> animation,
     Animation<double> secondaryAnimation,
   ) {
+    _FlippableTweenSequence<double> _closedOpacityTween;
+    _FlippableTweenSequence<double> _openOpacityTween;
+    _FlippableTweenSequence<Color> _colorTween;
+
+    switch (transitionType) {
```

Comment:
We could do this switching in the constructor and save the result in instance variables since the value of transitionType doesn't change for the duration of this object.

Reply:
I moved colorTween, openOpacityTween and closedOpacityTween to be defined in the constructor.